lib/spack/spack/cmd/mark.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from __future__ import print_function
+
+import spack.cmd
+import spack.error
+import spack.package
+import spack.cmd.common.arguments as arguments
+import spack.repo
+import spack.store
+from spack.database import InstallStatuses
+
+from llnl.util import tty
+
+description = "mark packages as explicitly or implicitly installed"
+section = "admin"
+level = "long"
+
+error_message = """You can either:
+    a) use a more specific spec, or
+    b) use `spack mark --all` to mark ALL matching specs.
+"""
+
+# Arguments for display_specs when we find ambiguity
+display_args = {
+    'long': True,
+    'show_flags': False,
+    'variants': False,
+    'indent': 4,
+}
+
+
+def setup_parser(subparser):
+    arguments.add_common_arguments(
+        subparser, ['installed_specs'])
+    subparser.add_argument(
+        '-e', '--explicit', action='store_true', dest='explicit',
+        help="Mark packages as explicitly installed.")
+    subparser.add_argument(
+        '-i', '--implicit', action='store_true', dest='implicit',
+        help="Mark packages as implicitly installed.")
```

Comment:
Can you use `subparser.add_mutually_exclusive_group()` for this

Reply:
Didn't know about that, thanks!